### PR TITLE
[exp][xdl] Check if expo SDK version is deprecated before build

### PR DESCRIPTION
**Pull request by jakubste**

### CODE REVIEW

File: packages/xdl/src/Versions.js [hunk 1]
```diff
@@ -109,18 +109,36 @@ export async function canTurtleBuildSdkVersion(sdkVersion, platform) {
...
+    );
   }
 
-  try {
-    const turtleSdkVersion = turtleSdkVersions[platform];
-    return semver.gte(turtleSdkVersion, sdkVersion);
-  } catch (e) {
+  const turtleSdkVersions = await Api.turtleSdkVersionsAsync();
+  const expoSdkVersions = await Api.sdkVersionsAsync();
+  const expoSdkVersion = _.values(
```
Comment (dsokal) [inline packages/xdl/src/Versions.js]: https://lodash.com/docs/4.17.10#chain

File: packages/xdl/src/Versions.js [hunk 2]
```diff
@@ -109,18 +109,36 @@ export async function canTurtleBuildSdkVersion(sdkVersion, platform) {
...
+    _.pickBy(
+      expoSdkVersions,
+      (props, version) => semver.major(version) === semver.major(sdkVersion)
```
Comment (dsokal) [inline packages/xdl/src/Versions.js]: Have we ever had more than one field for a given major SDK version in the `expoSdkVersions` object?

Comment (jakubste) [reply packages/xdl/src/Versions.js]: No.

Comment (dsokal) [reply packages/xdl/src/Versions.js]: So why do you need this complicated logic? Isn't `expoSdkVersions.find` function enough?

File: packages/xdl/src/Versions.js [hunk 3]
```diff
@@ -109,18 +109,35 @@ export async function canTurtleBuildSdkVersion(sdkVersion, platform) {
...
 
-  try {
-    const turtleSdkVersion = turtleSdkVersions[platform];
-    return semver.gte(turtleSdkVersion, sdkVersion);
-  } catch (e) {
+  const turtleSdkVersions = await Api.turtleSdkVersionsAsync();
+  const expoSdkVersions = await Api.sdkVersionsAsync();
+
+  const expoSdkVersion = _(expoSdkVersions)
+    .filter((props, version) => semver.major(version) === semver.major(sdkVersion))
```
Comment (dsokal) [inline packages/xdl/src/Versions.js]: `_.find`? :D

File: packages/xdl/src/Versions.js (hunk 2 above)
Comment (jakubste) [reply packages/xdl/src/Versions.js]: I am naive enough to thought we could use it someday.

**Pull request merged**